docs/querying/caching.md

```diff
@@ -22,63 +22,75 @@ title: "Query caching"
+
+- On Historicals, the default. Enable segment-level cache population on Historicals for larger production clusters to prevent Brokers from having to merge all query results. When you enable cache population on Historicals instead of Brokers, the Historicals merge their own local results and put less strain on the Brokers.
+
+- On ingestion tasks in the Peon or Indexer service. Larger production clusters should enable segment-level cache population on task execution services only to prevent Brokers from having to merge all query results. When you enable cache population on task execution services instead of Brokers, the the task execution services to merge their own local results and put less strain on the Brokers.
+
+     Task executor services only support caches that store data locally. For example the `caffeine` cache. This restriction exists because the cache stores results at the level of intermediate partial segments generated by the ingestion tasks. These intermediate partial segments may not be identical across task replicas. Therefore task executor services ignore remote cache types such as `memcached`.
+
+**Whole-query cache** is only available on Brokers.
+
+## Performance considerations for caching
+
+Consider the following as you develop your caching strategy:
+- Caching enables increased concurrency on the same system, therefore leading to noticeable performance improvements for queries on Druid clusters handling throughput for concurrent, mixed workloads.
+
+- If you are looking to improve response time for a single query or page load, you should ignore caching. In general, response time for a single task should meet performance objectives even when the cache is cold.
+
+- During query processing, the segment cache intercepts the query and sends the results directly to the Broker. This way the query bypasses the data server processing threads. For queries requiring minimal processing in the Broker, cached queries are very quick. If work done on the Broker causes a query bottleneck, enabling caching results in little noticeable query improvement.
+
+- The largest performance gains from segment caching tend to apply to `topN` and time series queries. The impact is less for `groupBy` queries.  The same applies to queries with or without joins.
+
+### Scenarios where caching does not increase query performance
+
+Caching does not solve all types of query peformance issues. For each cache type there are scenarios where caching is likely to be of little benefit.
+
+**Segment caching** doesn't work for TBD.
+
+**Whole-query caching** doesn't work for the following:
+- `GroupBy` v2 queries
+-  queries with joins
+-  queries on real-time data.
```

Comment:
I think we shouldn't mention this here. The other 2 in the list are specific limitations of the Druid system. This point talks about the fact that if you are querying real time data (last 5 minutes), you will never have a cached response since data for the last 5 minutes is always coming in. What do you think?
```suggestion
```